sys/arm64/arm64/efirt_support.S

```diff
@@ -70,11 +70,15 @@ ENTRY(efi_rt_arch_call_nofault)
 	ldr	c6, [c0, #EC_FPTR]
 
 	/* Jump to 0f + 4 * (5 - ec_argcnt) + 1 */
+#ifdef __ARM_MORELLO_PURECAP_BENCHMARK_ABI
+	adr	c9, 0f
+#else
 	adr	c9, 0f + 1
+#endif
 	mov	x10, #20
 	sub	x8, x10, x8, lsl #2
-	add	c9, c9, x8, uxtx
-	br	c9
+	add	x9, x9, x8, uxtx
```

Comment:
```suggestion
	add	c9, c9, x8, uxtx
```
Leave alone, only use the subregister for the branch itself, so as to make the workaround more limited in scope (also what Morello LLVM does)